Removal of onTNTExitPortal

TreemanKing opened the pull request:
~~Removes the entity preemptively rather than reactively.~~

Edit: Removes the event entirely.

Fixes #2119

TreemanKing committed:
Update EntityEventHandler.java

TreemanKing committed:
Update EntityEventHandler.java

RoboMWM commented on src/main/java/me/ryanhamshire/GriefPrevention/EntityEventHandler.java:
```diff
@@ -335,11 +335,11 @@ public void onFallingBlockEnterPortal(EntityPortalEnterEvent event)
         if (event.getEntityType() != EntityType.PRIMED_TNT)
             return;
-        if (event.getTo().getWorld().getEnvironment() != Environment.THE_END)
+        if (event.getLocation().getBlock().getType() == Material.END_PORTAL)
```
Would this handle cases of end portal blocks that don't actually end up teleporting an entity/are plugin-handled/end portals in non-default worlds?

TreemanKing replied:
Yes, will deal with all of the above. 

Using the EntityEnterPortalEvent, you can only determine the location of entry and what block it is. 

It shouldn't interfere with any portal plugins, however would probably be best to be configurable for future references if the server admins would like it to be close to the vanilla state.

Jikoo replied:
I think Robo's more asking "will this result in TNT that is not teleporting being removed" and I believe the answer is yes, though I'm not sure that it matters much. We could swap to the EntityTeleportEvent instead and see if it's from an end world to a non-end world with the correct cause.

TreemanKing replied:
Yes, it would although as you said I don't believe there are many problems that could arise as a result of this change. In most cases, if the End hasn't been generated, the TNT will just drop into the lava at the end portal structure.

In terms of clashes with other possible plugins which use portals, they would probably there own instances to remove other entities from teleporting.

I've attempted the EntityTeleportEvent before this and it would still crash the server.

Jikoo replied:
It crashes even when cancelled? That seems shocking, honestly. Either way, yeah, I don't see this implementation as being problematic.

TreemanKing replied:
Yeah, I tried that method first before attempting this one. Obviously the destination would be a lot better to use but it would also crash the server.

TreemanKing committed:
Merge branch 'fixPrimedTNTCrash' of https://github.com/TreemanKing/GriefPrevention into fixPrimedTNTCrash

TreemanKing committed:
Update EntityEventHandler.java - Add configuration options to BlockTNTTeleport

Added Configuration Option (set by default to false) to prevent teleport of TNT to the end.

TreemanKing committed:
Remove TNT End Portal Event

Retire Event - Redundant

RoboMWM committed:
Merge branch 'master' into fixPrimedTNTCrash
